gdal/ogr/ogrsf_frmts/mdb/ogrmdbjackcess.cpp

```diff
@@ -162,9 +167,41 @@ int OGRMDBJavaEnv::Init()
     {
         JavaVM* vmBuf[1];
         jsize nVMs;
+        int ret = 0;
+
+#if JVM_LIB_DLOPEN
+        const char *jvmLibPtr = "libjvm.so";
+        char jvmLib[PATH_MAX];
+
+        FILE *javaCmd = popen("\"${JAVA_HOME}${JAVA_HOME:+/bin/}java\" -XshowSettings 2>&1 | sed -n '/\\bsun\\.boot\\.library\\.path =/s:.* = \\(.*\\):\\1/server/libjvm.so:p'", "r");
```

Comment:
Yes, it makes me cringe a little too but I did try quite hard to think of a better way that doesn't involve a big chunk of C++, which isn't my strongest language.

It's POSIX-compliant shell code, which is as good as one can expect. That covers bash, dash, and ksh. I did try tcsh just for you but sadly it didn't work there. I gather MacOS defaults to bash and has sed available though I am unable to test its sed implementation. I'd be surprised if it didn't work.

There's only so far I'm willing to go for Windows and that's too far, sorry. I don't even have Visual Studio installed these days and I probably have a day's worth of Windows Updates waiting for me.